ground/gcs/src/plugins/coreplugin/iboardtype.cpp

```diff
@@ -11,41 +11,76 @@ QStringList IBoardType::queryChannelBanks()
 {
     QStringList banksStringList = QStringList();
 
-    foreach(QVector<qint32> channelBank, channelBanks) {
-        qint32 tmpChannel = -1;
-        qint32 lastWrittenChannel = -1;
+    foreach(QVector<qint32> channelBank, channelBanks) 
+	{
+        quint8 sequential = false;
         QString banksString;
 
-        for (int i=0; i<channelBank.size(); i++) {
-            quint8 channel = channelBank[i];
-            if (tmpChannel == -1) {
-                banksString.append(QString("%1").arg(channel));
-                lastWrittenChannel = channel;
-            } else if (channel == tmpChannel + 1) {
-                // At the end of the list, append the final number
-                if (i == channelBank.size()-1) {
-                    banksString.append(QString("-%1").arg(channel));
-                }
-            }
-            else {
-                if (lastWrittenChannel == tmpChannel) { // If we just wrote the tmpChannel value, write a comma and the new value
-                    banksString.append(QString(",%1").arg(channel));
-                }
-                else if (i < channelBank.size()-1) { // If this isn't the last element, add a comma
-                    banksString.append(QString("-%1,%2").arg(tmpChannel).arg(channel));
-                } else {
-                    banksString.append(QString("-%1").arg(channel));
-                }
-                lastWrittenChannel = channel;
-            }
-            tmpChannel = channel;
-        }
+		for (int i = 0; i < channelBank.size(); i++) 
+		{
+			if (i == 0)															 // First channel in bank
+			{
+				banksString.append(QString("%1").arg(channelBank[i]));
+				
+				if (channelBank.size() > 1)
+				{
```

Comment:
"{" moved above will mute now, but all other "if" and "else" braces should be moved to the same line